applications/nrf5340_audio/src/audio_module/src/aobj_api.h

```diff
@@ -0,0 +1,494 @@
+/*
+ * Copyright(c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef _AMOD_API_H_
+#define _AMOD_API_H_
+
+#include <zephyr/kernel.h>
+
+#include "data_fifo.h"
+#include "ablk_api.h"
+
+/**
+ * @brief Module type.
+ */
+enum amod_type {
+	/* The module type is undefined */
+	AMOD_TYPE_UNDEFINED = 0,
+
+	/* This is an input processing module */
+	AMOD_TYPE_INPUT,
+
+	/* This is an output processing module */
+	AMOD_TYPE_OUTPUT,
+
+	/* This is an input/output processing module */
+	AMOD_TYPE_IN_OUT
+};
+
+/**
+ * @brief Module state.
+ */
+enum amod_state {
+	/* Module state undefined */
+	AMOD_STATE_UNDEFINED = 0,
+
+	/* Module is in the configured state */
+	AMOD_STATE_CONFIGURED,
+
+	/* Module is in the running state */
+	AMOD_STATE_RUNNING,
+
+	/* Module is in the stopped state */
+	AMOD_STATE_STOPPED
+};
+
+/**
+ * @brief Module's private handle.
+ */
+struct handle;
+
+/**
+ * @brief Private context for the module.
+ */
+struct amod_context;
+
+/**
+ * @brief Module's private configuration.
+ */
+struct amod_configuration;
+
+/**
+ * @brief Private pointer to a module's functions.
+ */
+struct amod_functions {
+	/**
+	 * @brief Function for opening a module.
+	 *
+	 * @param handle         A handle to the module instance
+	 * @param configuration  A pointer to the module's current configuration
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*open)(struct handle *handle, struct amod_configuration *configuration);
+
+	/**
+	 * @brief Function to close an open module.
+	 *
+	 * @param handle  A handle to the module instance
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*close)(struct handle *handle);
+
+	/**
+	 * @brief Function to reconfigure a module.
+	 *
+	 * @param handle		 A handle to the module instance
+	 * @param configuration  Pointer to the module's configuration to set
+	 *
+	 * @return 0 if successful, error otherwise
+	 */
+	int (*configuration_set)(struct handle *handle, struct amod_configuration *configuration);
```

Comment:
Nitpick: To shorten, I think configuration can be abbreviated as config

Reply:
Can this stay like this as it has no effect on the code size or cycle count but makes it clear as to the purpose of the function.